test/pvp-arena.js

```diff
@@ -1807,4 +1805,330 @@ contract("PvpArena", (accounts) => {
+        character3ID = await createCharacterInPvpTier(accounts[2], 2, "222");
+        //this char will be in a different tier
+        character4ID = await createCharacterInPvpTier(accounts[2], 3, "222");
+        character5ID = await createCharacterInPvpTier(accounts[2], 2, "222");
+        character6ID = await createCharacterInPvpTier(accounts[1], 2, "222");
+        const characterTier = await pvpArena.getTopTierPlayers(character1ID, {
+          from: accounts[1],
+        });
+
+        expect(characterTier[0].toString()).to.equal(character1ID.toString());
+        expect(characterTier[3].toString()).to.equal(character5ID.toString());
+      });
+    });
+    describe("after the fight", () => {
+      it("update the ranks of both the winner and the loser and add/subtract points respectively", async () => {
+        const winningPoints = await pvpArena.winningPoints();
+        const losingPoints = await pvpArena.losingPoints();
+        weapon1ID = await helpers.createWeapon(
+          accounts[2],
+          "111",
+          helpers.elements.water,
+          {
+            weapons,
+          }
+        );
+        weapon2ID = await helpers.createWeapon(
+          accounts[1],
+          "111",
+          helpers.elements.fire,
+          {
+            weapons,
+          }
+        );
+        character1ID = await createCharacterInPvpTier(
+          accounts[1],
+          2,
+          "222",
+          weapon2ID
+        );
+        character2ID = await createCharacterInPvpTier(accounts[2], 2);
+        character3ID = await createCharacterInPvpTier(accounts[2], 2, "222");
+        character4ID = await createCharacterInPvpTier(
+          accounts[2],
+          2,
+          "222",
+          weapon1ID
+        );
+
+        await characters.setTrait(character4ID, helpers.elements.water, {
+          from: accounts[0],
+        });
+        await characters.setTrait(character1ID, helpers.elements.fire, {
+          from: accounts[0],
+        });
+        // we set the ranking points to determine the tier, char2 being the first, followed by char 1 and then char3
+        await pvpArena.setRankingPoints(character1ID, 35, {
+          from: accounts[0],
+        });
+        await pvpArena.setRankingPoints(character2ID, 34, {
+          from: accounts[0],
+        });
+        await pvpArena.setRankingPoints(character3ID, 33, {
+          from: accounts[0],
+        });
+        await pvpArena.setRankingPoints(character4ID, 32, {
+          from: accounts[0],
+        });
+
+        const winnerPreviousRankPoints =
+          await pvpArena.getCharacterRankingPoints(character4ID);
+        const loserPreviousRankPoints =
+          await pvpArena.getCharacterRankingPoints(character1ID);
+
+        await time.increase(await pvpArena.unattackableSeconds());
+        await pvpArena.requestOpponent(character4ID, {
+          from: accounts[2],
+        });
+
+        // perform a duel making sure character1 is always going to win
```

Comment:
How do you know `requestOpponent` will pick character1?

Reply:
Do you mean character 4?

Reply:
I meant character 4

Reply:
myb